job_controller/database/db_updater.py

```diff
@@ -174,9 +194,16 @@ def add_detected_run(
             reduction_inputs,
         )
         session = self.session_maker_func()
+        instrument = session.query(Instrument).filter_by(instrument_name=instrument_name).first()
+        if instrument is None:
+            instrument = Instrument(instrument_name=instrument_name)
```

Comment:
Without a unique constraint, will this create duplicate instruments?

Reply:
Uniqueness is ensured on the database side (I added it to the DB), but it will stop duplicates.

Reply:
If the DB prevents a duplicate, will sqlalchemy not raise something like an `IntegrityError` which would cause the rest of the method not to run?

Reply:
This code does stop duplicates, it functions correctly in the function below.